packages/shared/src/utils/platform/platform.js

```diff
@@ -12,18 +12,22 @@ export const isMT5 = () =>
     /^\/mt5/.test(window.location.pathname) ||
     (/^\/(br_)/.test(window.location.pathname) && window.location.pathname.split('/')[2] === 'mt5');
 
-export const getPlatformHeader = routing_history => {
-    if (isBot() || isNavigationFromPlatform(routing_history, routes.bot)) return 'DBot';
-    if (isMT5() || isNavigationFromPlatform(routing_history, routes.mt5)) return 'DMT5';
-    if (isNavigationFromPlatform(routing_history, routes.smarttrader)) return 'SmartTrader';
-    return 'DTrader';
-};
+export const isTrader = () =>
+    (/^\//.test(window.location.pathname) && window.location.pathname.length === 1) ||
+    (/^\/(br_)/.test(window.location.pathname) && window.location.pathname.split('/')[2] === '');
+
+// TODO: Replace platform names with `platform_name` enum after merging "carol/journal error notification"
+export const getPlatformInformation = routing_history => {
+    const should_be_bot = isBot() || (!isTrader() && !isMT5() && isNavigationFromPlatform(routing_history, routes.bot));
```

Comment:
- If `isBot()`, won't `isTrader()` and `isMT5()` automatically be false (so the checks are not required, as per code in upstream)?
- Could we use full on `if`-`else if` statements to make the code more clear?

```suggestion
if (isBot() || isNavigationFromPlatform(routing_history, routes.bot))) {
    return {
        header: 'DBot',
        icon: 'IcBrandDBot,
    }
} else if (isMT5() || isNavigationFromPlatform(routing_history, routes.mt5)) {
    return {
        header: 'DMT5',
        icon: 'IcBrandDmt5
    }
}
// Etc.
```

Reply:
@aaron-binary  For issue #1: The problem is that `routing_history` does not update correctly in case of routing between platforms with browsers routing buttons (`back` and `forward`). I used those conditions to handle that flaw. Else I should solve the issue with `routing_history`.

Reply:
What I mean is that if `isBot() === true`, `isTrader()` and `isMT5()` are both guaranteed to be `false`. The code for those functions is as per below:

```JS
export const isBot = () =>
    /^\/bot/.test(window.location.pathname) ||
    (/^\/(br_)/.test(window.location.pathname) && window.location.pathname.split('/')[2] === 'bot');

export const isMT5 = () =>
    /^\/mt5/.test(window.location.pathname) ||
    (/^\/(br_)/.test(window.location.pathname) && window.location.pathname.split('/')[2] === 'mt5');

export const isTrader = () =>
    (/^\//.test(window.location.pathname) && window.location.pathname.length === 1) ||
    (/^\/(br_)/.test(window.location.pathname) && window.location.pathname.split('/')[2] === '');
```

There is no way that `window.location.pathname.split('/')[2]` can be `bot`, `mt5`, or `''` at the same time, so the extra check for other platforms looks redundant.

Reply:
I used `!isTrader() && !isMT5()` for completing this part of the condition `isNavigationFromPlatform(routing_history, routes.bot)`. The first code was this `isBot() || isNavigationFromPlatform(routing_history, routes.bot)`. Now assume a condition which route to `Trader` from `Bot` with browser back button => `isBot()` equals to `false` but for `isNavigationFromPlatform` we will send `routing_history` with unupdated amount, and we will send `routes.bot` as constant so this one will return `true` and although `isBot()` is false because of the second condition, this condition wins which is false and I use those conditions for handeling the second part.
------------------------------------------
For `export const isTrader = () => !isBot() && !isMT5();` you are right, but I use `!isTrader()` which means `isBot() || isMt5()` so I had to check `isMt5()` as well.
------------------------------------------
Base on all of these I think it is better to fix the updating problem of `routing_history` and omit all of these added conditions. I think it gets me some time.